lightning/src/ln/channelmonitor.rs

```diff
@@ -1237,21 +1257,37 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 	/// is important that any clones of this channel monitor (including remote clones) by kept
 	/// up-to-date as our local commitment transaction is updated.
 	/// Panics if set_their_to_self_delay has never been called.
-	pub(super) fn provide_latest_local_commitment_tx_info(&mut self, commitment_tx: LocalCommitmentTransaction, local_keys: chan_utils::TxCreationKeys, feerate_per_kw: u64, htlc_outputs: Vec<(HTLCOutputInCommitment, Option<Signature>, Option<HTLCSource>)>) -> Result<(), MonitorUpdateError> {
+	pub(super) fn provide_latest_local_commitment_tx_info(&mut self, mut commitment_tx: LocalCommitmentTransaction, local_keys: chan_utils::TxCreationKeys, feerate_per_kw: u64, htlc_outputs: Vec<(HTLCOutputInCommitment, Option<Signature>, Option<HTLCSource>)>) -> Result<(), MonitorUpdateError> {
 		if self.their_to_self_delay.is_none() {
 			return Err(MonitorUpdateError("Got a local commitment tx info update before we'd set basic information about the channel"));
 		}
+		let txid = commitment_tx.txid();
+		self.current_local_commitment_number = 0xffff_ffff_ffff - ((((commitment_tx.without_valid_witness().input[0].sequence as u64 & 0xffffff) << 3*8) | (commitment_tx.without_valid_witness().lock_time as u64 & 0xffffff)) ^ self.commitment_transaction_number_obscure_factor);
```

Comment:
Wait, huh. I thought we were doing all the updates after the tx_handler.provide_latest_local_ call. Isn't this a severe bug - if the remote ChannelMonitor is updating we may end up with the wrong current_local_commitment_number set, breaking claiming?

Reply:
It's not nasty as this, because it would mean we already have a lockdown ChannelMonitor and a local commitment transaction broadcast, as far as I can tell it would just mean an already-closed channel wouldn't be call for force-closure a second time by ChannelManager deserialization. But yes that's clearly a bug so fixed it